python-spec/src/somacore/query/query.py

```diff
@@ -341,8 +356,15 @@ def _read(
             for _xname in all_x_arrays
         }
 
+        # TODO: do it in parallel?
+        obsm = dict()
+        for key in obsm_keys:
+            obsm[key] = self._axism_inner_csr(_Axis.OBS, key)
```

Comment:
why is this a CSR - it should be an ndarray... 

See [AnnData obsm doc](https://anndata.readthedocs.io/en/latest/generated/anndata.AnnData.obsm.html#anndata.AnnData.obsm)